### PR TITLE
Add full scanning, auth/ssl/license prompts

### PR DESCRIPTION
**Pull request by tnederlof**

### CODE REVIEW

**Review by dpastoor — CHANGES_REQUESTED** [review]
This is great work so far, i have a number of comments inline, however the final piece to comment on your error handling in some places.

In particular, you shouldn't really log.Fatal anywhere until the top level of the app (or wherever you really want to handle it. Eg you have areas where you have

if err != nil {
log.Fatal(err)
}

where instead you should just return the error and let it bubble up. Then at the top level of your app you can chose how to handle it, eg there log.Fatal

Now lets see it in action :-)

File: internal/config/config.go [hunk 1]
```diff
@@ -1,15 +1,27 @@
 package config
 
+type AuthType int
+
+const (
```
[the remark this answers is not in the document]
Comment (tnederlof) [reply internal/config/config.go:5]: This is interesting, I was thinking about that as often you want to return a zero value, not sure how to handle that in this case. In some ways its more confusing then it helps but ill dig in more.

**Comment by tnederlof** [comment]
Thanks for the detailed comments @dpastoor learning rapidly. I see what you mean about error handling, I was struggling what to do there. Just went through an overhaul, let me know what you think about the error handling now.

**Comment by tnederlof** [comment]
Video of current functionalities

https://user-images.githubusercontent.com/180123/217387213-19fa1820-557d-4b86-9d6b-e98c29e1e338.mp4

File: internal/languages/R.go [hunk 2]
```diff
@@ -43,22 +41,43 @@ func isRDir(path string) (string, bool) {
...
 	if len(rVersions) == 0 {
-		// TODO: if no R version found, send link to R installation doc
-		log.Fatal("no R versions found at locations: \n", strings.Join(GetRRootDirs(), "\n"))
+		fmt.Println("To install versions of R, please follow the instructions outline here: https://docs.posit.co/resources/install-r/")
+
+		errorMessage := "no R versions found at locations: \n" + strings.Join(GetRRootDirs(), "\n")
+		return []string{}, errors.New(errorMessage)
+	}
+
+	var anyOptLocations = make([]string, 0)
```
Comment (dpastoor) [inline internal/languages/R.go:56]: can just define var anyOptLocations []string or just anyOptLocations := []string{}

File: internal/languages/R.go [hunk 3]
```diff
@@ -88,10 +107,9 @@ func ScanForRVersions() ([]string, error) {
 		}
 
 	}
-	maybeR := which.Which("R")
-	if maybeR != "" {
-		foundVersions = append(foundVersions, maybeR)
-	}
+
+	maybeR, _ := exec.LookPath("R")
```
Comment (dpastoor) [inline internal/languages/R.go:111]: should capture the error and only add maybeR if err != nil since it will return an error if lookpath doesn't find it

**Review by dpastoor — COMMENTED** [review]
Great stuff - minor knits in review. Home much further do we want to go in this branch? We discussed some potentially architectural redesigns today around how data is stored as we move closer to doing something with this data, but I feel like that would be worth doing after we get some PoC stopping point and merge down so the overall changesets can get more targeted.

One area to put on the todo is considering the cmd.Output() which makes you wait to collect all the output into a buffer. We could switch this to actually run the command and pipe the output as it runs such that users can see progress interactively _or_ if we don't want to see the scroll of stuff across their screen we can at least start a spinner before calling cmd.Output() then stop the spinner after so they know something is going on

**Comment by tnederlof** [comment]
@dpastoor thanks as always, learning rapidly with your feedback.

I made the changes you pointed out. This was a massive PR since it is early days and had to refactor the errors, etc. Lets merge this one and then going forward can do smaller targeted PRs for certain features/aspects on the TODO list.

Next on the list will be outputting config information somehow (print to screen, in actually files, etc), either way creating a layer to map values from the struct to their actual config locations will make this useful and able to start being tested/POCed.

**Review by dpastoor — APPROVED** [review]

**Pull request merged**